tests/framework/db/mssql/QueryBuilderTest.php

```diff
@@ -67,36 +67,178 @@ public function testOffset()
+            'objname' => $column,
+            'name' => 'MS_description',
+            'value' => $columnComment2,
+        ], $result[0]);
+    }
 
-        $expected = "sp_dropextendedproperty @name = N'MS_Description', @level1type = N'Table',  @level1name = comment";
-        $sql = $qb->dropCommentFromTable('comment');
-        $this->assertEquals($expected, $sql);
+    public function testCommentAdditionOnQuotedTableOrColumn()
+    {
+        $table = 'stranger \'table';
+        $tableComment = 'A comment for stranger \'table.';
+        $this->runAddCommentOnTable($tableComment, $table);
+        $resultTable = $this->getCommmentsFromTable($table);
+        $this->assertEquals([
+            'objtype' => 'TABLE',
+            'objname' => $table,
+            'name' => 'MS_description',
+            'value' => $tableComment,
+        ], $resultTable[0]);
+
+        $column = 'stranger \'field';
+        $columnComment = 'A comment for stranger \'field column in stranger \'table.';
+        $this->runAddCommentOnColumn($columnComment, $table, $column);
+        $resultColumn = $this->getCommentsFromColumn($table, $column);
+        $this->assertEquals([
+            'objtype' => 'COLUMN',
+            'objname' => $column,
+            'name' => 'MS_description',
+            'value' => $columnComment,
+        ], $resultColumn[0]);
+    }
+
+    public function testCommentRemovalFromTableAndFromColumn()
+    {
+        $table = 'profile';
+        $tableComment = 'A comment for profile table.';
+        $this->runAddCommentOnTable($tableComment, $table);
+        $this->runDropCommentFromTable($table);
+        $result = $this->getCommmentsFromTable($table);
+        $this->assertEquals([], $result);
+
+        $column = 'description';
+        $columnComment = 'A comment for description column in profile table.';
+        $this->runAddCommentOnColumn($columnComment, $table, $column);
+        $this->runDropCommentFromColumn($table, $column);
+        $result = $this->getCommentsFromColumn($table, $column);
+        $this->assertEquals([], $result);
+    }
+
+    public function testCommentRemovalFromQuotedTableOrColumn()
+    {
+        $table = 'stranger \'table';
+        $tableComment = 'A comment for stranger \'table.';
+        $this->runAddCommentOnTable($tableComment, $table);
+        $this->runDropCommentFromTable($table);
+        $result = $this->getCommmentsFromTable($table);
+        $this->assertEquals([], $result);
+
+        $column = 'stranger \'field';
+        $columnComment = 'A comment for stranger \'field in stranger \'table.';
+        $this->runAddCommentOnColumn($columnComment, $table, $column);
+        $this->runDropCommentFromColumn($table, $column);
+        $result = $this->getCommentsFromColumn($table, $column);
+        $this->assertEquals([], $result);
+    }
+
+    public function testCommentColumn()
+    {
+        $this->markTestSkipped("Testing the behavior, not sql generation anymore.");
```

Comment:
What’s wrong with testing generated SQL? It’s faster than running those queries.

Reply:
1) Originally we were testing the sql, the tests were passing but the comments was not working. Yeah, it was a test error, but just in case...
2) In dealing with some MS products that have theirs own ~~strange~~ "particular" way to do the things, It seems safer to test the real thing happening instead of only sql code. By the way, the additional time seems derisory.